src/stackwalk.c

```diff
@@ -122,15 +142,25 @@ int jl_unw_stepn(bt_cursor_t *cursor, jl_bt_element_t *bt_data, size_t *bt_size,
                 call_ip = 0;
             }
             jl_bt_element_t *bt_entry = bt_data + n;
-            size_t entry_sz = 0;
-            if (add_interp_frames && jl_is_enter_interpreter_frame(call_ip) &&
-                (entry_sz = jl_capture_interp_frame(bt_entry, thesp, thefp, maxsize-n)) != 0) {
-                n += entry_sz;
-            } else {
-                bt_entry->uintptr = call_ip;
-                n++;
+            jl_gcframe_t *pgcstack;
+            if ((pgcstack = is_enter_interpreter_frame(ppgcstack, thesp))) {
+                size_t add = jl_capture_interp_frame(bt_entry, pgcstack, maxsize - n);
+                n += add;
+                bt_entry += add;
+                while ((pgcstack = is_enter_interpreter_frame(ppgcstack, thesp))) {
+                    // If the compiler got inlining-happy, or the user tried to
+                    // push multiple frames (or the unwinder got very
+                    // confused), we could end up here. That doesn't happen
+                    // now, so just ignore this possibility. If we want this,
+                    // we can work on adding support for it later.
+                }
             }
+            bt_entry->uintptr = call_ip;
+            n++;
         }
+        // TODO: if we have some pgcstack entries remaining (because the
+        // unwinder failed and returned !have_more_frames early), we could
+        // still append those frames here
```

Comment:
Am I correct in thinking that this would, however, lead to missing all compiled frames which interleave with the frames which happen to be interpreted? That sounds like it would be more confusing than helpful for the average user.

Reply:
Yeah, it'd just jump to the frames it could still find. Happens sometimes anyways if the unwinder fails.